proposals/0298-asyncsequence.md

```diff
@@ -194,12 +195,17 @@ If an `AsyncIteratorProtocol` type has cleanup to do upon cancellation, it can d
 
 1. After checking for cancellation using the `Task` API.
 2. In its `deinit` (if it is a class type).
+
 ### Rethrows
 
 This proposal will take advantage of a separate proposal to add specialized `rethrows` conformance in a protocol, pitched [here](https://forums.swift.org/t/pitch-rethrowing-protocol-conformances/42373). With the changes proposed there for `rethrows`, it will not be required to use `try` when iterating an `AsyncSequence` which does not itself throw.
 
 The `await` is always required because the definition of the protocol is that it is always asynchronous.
 
+### End of Iteration
+
+After an `AsyncIteratorProtocol` types returns `nil` or throws an error from its `next()` method, all future calls to `next()` must return `nil`. This matches the behavior of `IteratorProtocol` types and is important, since calling an iterator's `next()` method is the only way to determine whether iteration has finished.
```

Comment:
<https://forums.swift.org/t/pitch-clarify-end-of-iteration-behavior-for-asyncsequence/45548/7>

```suggestion
After an `AsyncIteratorProtocol` type returns `nil` or throws an error from its `next()` method, all future calls to `next()` must return `nil` or throw an error. This matches the behavior of `IteratorProtocol` types and is important, since calling an iterator's `next()` method is the only way to determine whether iteration has finished.
```